stock_barcodes/wizard/stock_barcodes_read.py

```diff
@@ -0,0 +1,184 @@
+            return
+        packaging = self.env['product.packaging'].search(domain)
+        if packaging:
+            if len(packaging) > 1:
+                self._set_messagge_info(
+                    'more_match', _('More than one package found'))
+                return
+            self.action_packaging_scaned_post(packaging)
+            self.action_done()
+            return
+        lot = self.env['stock.production.lot'].search([
+            ('name', '=', barcode),
+            ('product_id', '=', self.product_id.id),
+        ])
+        if lot:
+            self.lot_id = lot
+            return
+        location = self.env['stock.location'].search(domain)
+        if location:
+            self.location_id = location
+            return
+        self._set_messagge_info('not_found', _('Barcode not found'))
+
+    def _barcode_domain(self, barcode):
+        return [('barcode', '=', barcode)]
+
+    def on_barcode_scanned(self, barcode):
+        self.barcode = barcode
+        self.reset_qty()
+        self.process_barcode(barcode)
+
+    def check_done_conditions(self):
+        if not self.product_qty:
+            self._set_messagge_info('info', _('Waiting quantities'))
+            return False
+        if self.manual_entry:
+            self._set_messagge_info('success', _('Manual entry OK'))
+        return True
+
+    def action_done(self):
+        if not self.check_done_conditions():
+            return False
+        self._add_read_log()
+        return True
+
+    def action_product_scaned_post(self, product):
+        self.packaging_id = False
+        if self.product_id != product:
+            self.lot_id = False
+        self.product_id = product
+        self.product_qty = 0.0 if self.manual_entry else 1.0
+
+    def action_packaging_scaned_post(self, packaging):
+        self.packaging_id = packaging
+        if self.product_id != packaging.product_id:
+            self.lot_id = False
+        self.product_id = packaging.product_id
+        self.packaging_qty = 0.0 if self.manual_entry else 1.0
+        self.product_qty = packaging.qty * self.packaging_qty
+
+    def action_clean_lot(self):
+        self.lot_id = False
+
+    def action_manual_entry(self):
+        return True
+
+    def _prepare_scan_log_values(self):
+        return {
+            'name': self.barcode,
+            'location_id': self.location_id.id,
+            'product_id': self.product_id.id,
+            'packaging_id': self.packaging_id.id,
+            'lot_id': self.lot_id.id,
+            'packaging_qty': self.packaging_qty,
+            'product_qty': self.product_qty,
+            'manual_entry': self.manual_entry,
+            'res_model_id': self.res_model_id.id,
+            'res_id': self.res_id,
+        }
+
+    def _add_read_log(self):
+        if self.product_qty:
+            vals = self._prepare_scan_log_values()
+            self.env['stock.barcodes.read.log'].create(vals)
+
+    @api.depends('_barcode_scanned')
```

Comment:
where this value is defined ? propose to replace by 'product_id', 'location_id'. (and to be coherent : 'res_model_id', 'res_id' also)

Reply:
Hi, the Odoo core module barcodes add this field.
This allow listen barcode scanned from hardware scanner.
https://github.com/OCA/OCB/blob/11.0/addons/barcodes/models/barcode_events_mixin.py#L14
Anyway I change this for product_id only, when a location is changed the product_id field is set to False